contracts/transfer/tests/transfer.rs

```diff
@@ -62,40 +62,46 @@ fn instantiate<Rng: RngCore + CryptoRng>(
         "../../../target/wasm32-unknown-unknown/release/alice.wasm"
     );
 
-    let mut session = vm.genesis_session();
-
+    let mut session = rusk_abi::new_genesis_session(vm);
     session.set_point_limit(POINT_LIMIT);
-    rusk_abi::set_block_height(&mut session, 0);
 
     session
-        .deploy_with_id(rusk_abi::transfer_module(), transfer_bytecode)
+        .deploy(
+            transfer_bytecode,
+            ContractData::builder(OWNER).contract_id(TRANSFER_CONTRACT),
+        )
         .expect("Deploying the transfer contract should succeed");
 
     session
-        .deploy_with_id(ALICE_ID, alice_bytecode)
+        .deploy(
+            alice_bytecode,
+            ContractData::builder(OWNER).contract_id(ALICE_ID),
+        )
         .expect("Deploying the alice contract should succeed");
 
     session
-        .deploy_with_id(BOB_ID, bob_bytecode)
+        .deploy(
+            bob_bytecode,
+            ContractData::builder(OWNER).contract_id(BOB_ID),
+        )
         .expect("Deploying the bob contract should succeed");
 
     let genesis_note = Note::transparent(rng, psk, GENESIS_VALUE);
 
     // push genesis note to the contract
     let _: Note = session
-        .transact(
-            rusk_abi::transfer_module(),
-            "push_note",
-            &(0u64, genesis_note),
-        )
+        .call(TRANSFER_CONTRACT, "push_note", &(0u64, genesis_note))
         .expect("Pushing genesis note should succeed");
 
     let _: BlsScalar = session
-        .transact(rusk_abi::transfer_module(), "update_root", &())
+        .call(TRANSFER_CONTRACT, "update_root", &())
         .expect("Updating the root should succeed");
 
     // sets the block height for all subsequent operations to 1
-    rusk_abi::set_block_height(&mut session, 1);
+    let base = session.commit().expect("Committing should succeed");
+    let mut session = rusk_abi::new_session(vm, base, 1)
+        .expect("Instantiating new session should succeed");
+    session.set_point_limit(POINT_LIMIT);
```

Comment:
This is not clear to me.
Previous version didn't require to commit before return. Is something changed internally with `0.5`?

Reply:
Yes. We can't set session data - such as block height - after the session has started.